index.bs

```diff
@@ -387,233 +507,172 @@ Each \[[Stage]] value indicates the currently pending operation. If the \[[Resul
-1. Let _stage_ be _currentStage_.[[Stage]].
-1. Let _result_ be CreateObject().
-1. Call SimpleDefine(_result_, "stage", _stage_).
-1. Call SimpleDefine(_result_, "result", _result_).
-1. If _stage_ is "ready" then let _module_ be _entry_.[[Module]].
-1. Else let _module_ be *undefined*.
-1. Call SimpleDefine(_result_, "module", _module_).
-1. If _entry_.[[Error]] is *nothing*, then:
-  1. Call SimpleDefine(_result_, "error", *null*).
-1. Else:
-  1. Let _opt_ be CreateObject().
-  1. Call SimpleDefine(_opt_, "value", _entry_.[[Error]]).
-  1. Call SimpleDefine(_result_, "error", _opt_).
-1. Return _result_.
-</emu-alg>
-
-A <dfn id="registry-entry">registry entry</dfn> is a record with the following fields:
-
-<table>
-  <thead>
-    <tr>
-      <th>Internal Slot</th>
-      <th>Value Type (<em>non-normative</em>)</th>
-      <th>Description (<em>non-normative</em>)</th>
-    </tr>
-  </thead>
-  <tr>
-    <td>\[[Key]]</td>
-    <td>String</td>
-    <td>The resolved module key.</td>
-  </tr>
-  <tr>
-    <td>\[[Pipeline]]</td>
-    <td>A List</td>
-    <td>A list whose elements are stage records.</td>
-  </tr>
-  <tr>
-    <td>\[[Metadata]]</td>
-    <td>Object or <code>undefined</code></td>
-    <td>The metadata object passed through the pipeline.</td>
-  </tr>
-  <tr>
-    <td>\[[Dependencies]]</td>
-    <td>List of pairs of String and (<a>registry entry</a> or <code>undefined</code>), or <code>undefined</code>.</td>
-    <td>Table mapping unresolved names to their resolved modules.</td>
-  </tr>
-  <tr>
-    <td>\[[Module]]</td>
-    <td>Module Record or Function object or <code>undefined</code></td>
-    <td>The Module Record if the entry is ready, or a thunk if the entry is delayed; otherwise <code>undefined</code>.</td>
-  </tr>
-  <tr>
-    <td>\[[Error]]</td>
-    <td>Any or <b>nothing</b></td>
-    <td>An error that was encountered during one of the phases of the loading pipeline; <b>nothing</b> if no error has been encountered.</td>
-  </tr>
-</table>
-
-<h3 id="registry-constructor">The Registry Constructor</h3>
-
-The Registry constructor is the %Registry% intrinsic object. When called as a constructor it creates and initializes a new %Registry% object. %Registry% is not intended to be called as a function and will throw an exception when called in that manner.
-
-<h4 id="new-registry" aoid="Registry">Registry(loader)</h4>
+<h4 id="new-registry-entry" aoid="RegistryEntry">RegistryEntry(loader)</h4>
 
 When Registry is called with argument <i>loader</i>, the following steps are taken:
 
 <emu-alg>
 1. If NewTarget is *undefined*, then throw a *TypeError* exception.
 1. If Type(_loader_) is not Object, throw a *TypeError* exception.
-1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "%RegistryPrototype%", «[[RegistryData]]» ).
+1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "%RegistryPrototype%", «[[Loader]], [[Pipeline]], [[Module]], [[Metadata]], [[Dependencies]], [[Error]]» ).
```

Comment:
I think we should allow an optional second argument, which is required to be a valid module object if it is not **undefined**. This would be the way to synchronously circumvent the asynchronous loading process. So you can synchronously create a module status object for a completely loaded module via `new System.ModuleStatus(loader, module)`, and it will fast-forward directly to the ready state; otherwise it will start out in the initial state and can be advanced asynchronously via `.resolve` and `.reject`.